### PR TITLE
Fix docker build issue

### PR DESCRIPTION
**Pull request by lanluo-nvidia**
# Description

Fixes # (issue)
 ERROR: /workspace/torch_tensorrt/src/WORKSPACE:82:10: //external:devtools_deps: no such attribute 'requirements' in 'pip_repository' rule
#30 1483.6       ERROR: Error computing the main repository mapping: Encountered error while reading extension file 'requirements.bzl': no such package '@@devtools_deps//': error loading package 'external': Could not load //external package

Fix the failed docker build.

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project (You can use the linters)
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

File: docker/WORKSPACE.docker [hunk 1]
```diff
@@ -81,7 +81,7 @@ load("@rules_python//python:pip.bzl", "pip_parse")
 
 pip_parse(
     name = "devtools_deps",
-    requirements = "//:requirements-dev.txt",
+    requirements_lock = "//:requirements-dev.txt",
```
Comment (narendasan) [inline docker/WORKSPACE.docker:84]: Can we just use the MODULE.bazel system?

File: py/requirements.txt [hunk 2]
```diff
@@ -2,7 +2,23 @@ numpy
...
+#30 107.3      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 991.8/991.8 MB 3.4 MB/s eta 0:00:00
+#30 107.8   Downloading https://download.pytorch.org/whl/nightly/cu124/torch-2.5.0.dev20240805%2Bcu124-cp310-cp310-linux_x86_64.whl (992.7 MB)
+#30 114.6      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 992.7/992.7 MB 3.5 MB/s eta 0:00:00
+#30 115.2   Downloading https://download.pytorch.org/whl/nightly/cu124/torch-2.5.0.dev20240804%2Bcu124-cp310-cp310-linux_x86_64.whl (992.7 MB)
+#30 121.9      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 992.7/992.7 MB 3.9 MB/s eta 0:00:00
+#30 122.4   Downloading https://download.pytorch.org/whl/nightly/cu124/torch-2.5.0.dev20240803%2Bcu124-cp310-cp310-linux_x86_64.whl (992.7 MB)
+#30 129.2      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 992.7/992.7 MB 3.9 MB/s eta 0:00:00
+#30 129.8   Downloading https://download.pytorch.org/whl/nightly/cu124/torch-2.5.0.dev20240802%2Bcu124-cp310-cp310-linux_x86_64.whl (992.7 MB)
+torch>=2.5.0.dev20240617,<2.5.0.dev20240618
 torchvision>=0.19.0.dev,<0.20.0
```
Comment (narendasan) [inline py/requirements.txt:22]: Can we just skip torchvision as a dep?

File: docker/WORKSPACE.docker (hunk 1 above)
Comment (lanluo-nvidia) [reply docker/WORKSPACE.docker]: changed to MODULE.bazel

File: py/requirements.txt (hunk 2 above)
Comment (lanluo-nvidia) [reply py/requirements.txt:7]: This PR will only focus on the workspace file to MODULE.bazel change.
Will do this in a separate PR:https://github.com/pytorch/TensorRT/pull/3076

Comment (lanluo-nvidia) [reply py/requirements.txt:7]: if torchvision is not defined as dependency here, in the test it will still need to be pulled as dependency.
And mixing incompatible torch and torchvision is causing other errors, see pipeline error from PR:https://github.com/pytorch/TensorRT/pull/3076

File: py/requirements.txt [hunk 3]
```diff
@@ -2,7 +2,8 @@ numpy
 packaging
 pybind11==2.6.2
 --extra-index-url https://download.pytorch.org/whl/nightly/cu124
-torch>=2.5.0.dev,<2.6.0
+# it is trying to download all the torch versions from 0808 back to 0617 to find the matching torchvision 0.19.0 which cause disk space to be full
+torch>=2.5.0.dev20240617,<2.5.0.dev20240618
```
Comment (narendasan) [inline py/requirements.txt:6]: Can we get rid of this file and just use pyproject.toml?

File: docker/MODULE.bazel.docker [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+# NOTE: In the case you are using just the pre-cxx11-abi path or just the cxx11 abi path
+# with your local libtorch, just point deps at the same path to satisfy bazel.
+
+# NOTE: NVIDIA's aarch64 PyTorch (python) wheel file uses the CXX11 ABI unlike PyTorch's standard
+# x86_64 python distribution. If using NVIDIA's version just point to the root of the package
+# for both versions here and do not use --config=pre-cxx11-abi
+
+new_local_repository(
+    name = "libtorch",
+    path = "/opt/python3/site-packages/torch/",
```
Comment (narendasan) [inline docker/MODULE.bazel.docker:52]: For our containers could we not just use libtorch-latest or pip install torch nightly and use the same system as packaging/pre_build to fill in the path to torch?

Comment (lanluo-nvidia) [reply docker/MODULE.bazel.docker:52]: for container build, we first have Module.bazel and then do 
`python -m pip wheel .  --extra-index-url https://download.pytorch.org/whl/nightly/cu124 -w dist`
to install torch_tensorrt and dependencies like torch.

for windows/linux build, we first install torch and then build torch_tensorrt via Module.bazel so we can use the packaging/pre_build to fill in the path to torch
`export TORCH_INSTALL_PATH=$(python -c "import torch, os; print(os.path.dirname(torch.__file__))")`

File: py/requirements.txt (hunk 3 above)
Comment (lanluo-nvidia) [reply py/requirements.txt:6]: I prefer change this in a separate PR, because it might need to make changes on the windows/linux build/test/release scripts.
So that this PR can be focused on the docker change only.

Comment (lanluo-nvidia) [reply py/requirements.txt:6]: limit torch version to torch>=2.5.0.dev20240617,<2.5.0.dev20240618 is not ideal, it should be temporary.
we should try to find a solution of bump torchvision version to 0.20.0
currently if bump torchvision to 0.20.0 we have failed testcases: 
Some arguments (weight, bias, running_mean, running_var) to batch_norm are passed with different types between torchvision 0.19 and 0.20 in Test torch compile backend CI.
https://github.com/pytorch/TensorRT/actions/runs/10224911783/job/28582163503?pr=3001

**Review by narendasan — APPROVED** [review]
LGTM
